src/privatesend/privatesend-client.cpp

```diff
@@ -1168,9 +1168,9 @@ bool CPrivateSendClientSession::StartNewQueue(CAmount nBalanceNeedsAnonymized, C
 
         // try to get a single random denom out of setAmounts
         while (nSessionDenom == 0) {
-            for (const auto& amount : setAmounts) {
+            for (auto it = setAmounts.rbegin(); it != setAmounts.rend(); ++it) {
                 if (setAmounts.size() > 1 && GetRandInt(2)) continue;
-                nSessionDenom = CPrivateSend::AmountToDenomination(amount);
+                nSessionDenom = CPrivateSend::AmountToDenomination(*it);
```

Comment:
Can you explain the reasoning of "Prefer larger denoms when starting new mixing queues" and the issue you're referencing?

Reply:
That's how PS used to work before that refactoring actually because it was a vector wich was filed from largest to smallest denoms, so this simply brings back the original behaviour.